### PR TITLE
Add svm bid status

### PR DESCRIPTION
**Pull request by danimhr**
This PR aims to add svm bid status type

### CODE REVIEW

File: auction-server/src/state.rs [hunk 1]
```diff
@@ -885,9 +1189,17 @@ impl Store {
         match auction.tx_hash {
             Some(tx_hash) => bids
                 .into_iter()
-                .filter(|bid| match bid.get_core_fields().status {
-                    BidStatus::Submitted { result, .. } => result == tx_hash,
-                    _ => false,
+                .filter(|bid| match bid {
+                    SimulatedBid::Evm(bid) => match bid.core_fields.status {
+                        BidStatusEvm::Submitted { result, .. } => {
+                            result.as_bytes().to_vec() == tx_hash
```
Comment (m30m) [inline auction-server/src/state.rs:1195]: use convert_tx_hash

File: auction-server/src/state.rs [hunk 2]
```diff
@@ -322,50 +503,232 @@ pub enum BidStatus {
...
+}
+
+pub trait BidStatusTrait:
+    Clone
+    + Into<models::BidStatus>
+    + std::fmt::Debug
+    + TryFrom<(models::Bid, Option<models::Auction>)>
+    + Into<BidStatus>
+{
+    type TxHash: Clone + std::fmt::Debug;
```
Comment (m30m) [inline auction-server/src/state.rs:590]: you can do something like this:
```
type TxHash: Clone + std::fmt::Debug + AsRef<[u8]>;
    fn convert_tx_hash(tx_hash: &Self::TxHash) -> Vec<u8> {
        tx_hash.as_ref().to_vec()
    }
```

File: auction-server/src/state.rs [hunk 3]
```diff
@@ -434,32 +789,15 @@ impl SimulatedBidCoreFields {
...
-            SimulatedBid::Evm(bid) => SimulatedBid::Evm(SimulatedBidEvm { core_fields, ..bid }),
-            SimulatedBid::Svm(bid) => SimulatedBid::Svm(SimulatedBidSvm { core_fields, ..bid }),
+            SimulatedBid::Evm(bid) => bid.core_fields.id,
+            SimulatedBid::Svm(bid) => bid.core_fields.id,
         }
     }
 }
 
-impl TryFrom<(models::Bid, Option<models::Auction>)> for BidStatus {
+impl TryFrom<(models::Bid, Option<models::Auction>)> for BidStatusEvm {
```
Comment (m30m) [inline auction-server/src/state.rs:800]: I'd rather we use From and TryFrom traits only for conversions that are simple.
We can put these in dedicated functions inside the traits.

it's not easy to discover `let a: BidStatusEvm = (bid, auction).try_into()` is doable

File: auction-server/src/auction.rs [hunk 4]
```diff
@@ -379,18 +388,22 @@ async fn broadcast_lost_bids<T: SimulatedBidTrait>(
...
-                    BidStatus::Lost {
-                        result: tx_hash,
-                        index:  None,
-                    },
-                    auction,
-                )
-                .await
-            {
-                tracing::error!("Failed to broadcast bid status: {:?} - bid: {:?}", err, bid);
+            match T::get_bid_status(models::BidStatus::Submitted, None, tx_hash) {
```
[the remark this answers is not in the document]
Comment (danimhr) [reply auction-server/src/auction.rs:391]: Wow! good catch

File: auction-server/src/state.rs (hunk 3 above)
Comment (danimhr) [reply auction-server/src/state.rs:800]: It's better to put them inside a function.
The other reason for this is you cannot get the try_from error type when implementing a try_from for a trait.

**Review by m30m — APPROVED** [review]

**Pull request merged**